### PR TITLE
[#331] Add an option to choose the RSpec otherwise skip the Rspec configuration

### LINKED ISSUE #331: Add an option to choose the RSpec otherwise skip the Rspec configuration
## Why

It could be a requirement or just a hobby project where you want to learn more about the default rails test and its feature. In that case, it would be great to skip the RSpec configuration in the project.
 
## Who Benefits?

Everyone who wants to stick with rails default-test library.

### PR DESCRIPTION
**Pull request by przbadu**
- Close #331 

## What happened

Allow user to choose `Rspec` test framework, if `N` is chosen, stick with Rails default test framework.

## Insight

Make rspec Optional.

### CODE REVIEW

**Comment by przbadu** [comment]
@malparty Hi, I made changes to install RSpec as a testing library conditionally. Am I missing any RSpec dependencies that need to be conditionally loaded along with RSpec?

**Comment by przbadu** [comment]
@malparty Hi, I made changes to install RSpec as a testing library conditionally. Am I missing any RSpec dependencies that need to be conditionally loaded along with RSpec?

**Comment by malparty** [comment]
@przbadu hi, 
Thanks for your contribution :) I will test it tomorrow and also discuss it with some teammates so they can support me in checking that more in details 🙌

File: template.rb [hunk 1]
```diff
@@ -67,6 +65,10 @@ def apply_template!(template_root)
   post_default_addons_install
 
   # Add-ons - [Optional]
+  if yes?(install_addon_prompt('Rspec'))
+    delete_test_folder
+    apply '.template/addons/rspec/template.rb'
+  end
```
Comment (malparty) [inline template.rb:71]: In the case the addon is **not** selected, should we also remove the following?
- `./bin/test.sh` — as it depends on the `rspec` package
- `./docker-compose.test.yml` — as it is using the `test.sh` script

**Comment by malparty** [comment]
@przbadu there are some concerns I am not sure how to resolve at the moment:
- Readme.md: The test section provide rspec-specific instructions. Should we just add an indication that this section only applies when the `rspec` add-on has been selected? 🤔 
- `.rubocop.yml`: When the rspec add-on is not selected, we should remove line 3 because it requires a gem that is not installed (rubocop-rspec). Same for the last rules at the end of the file.

Let me know if you have an idea to handle them 🙌 

PS: We have some teammates off this week and next week so the team supporting our Rail Template might be slower than usual to review/merge PR, sorry for the inconvenience~

File: .template/addons/rspec/Gemfile.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+############################
+# Group: :development, :test
+############################
+insert_into_file 'Gemfile', after: /gem 'fabrication'.*\n/ do
+  <<~RUBY.indent(2)
+    # Testing
```
Comment (malparty) [inline .template/addons/rspec/Gemfile.rb:6]:
```suggestion
  <<~RUBY.indent(2)
  
    # Testing
```

We can add an empty line here to make it consitent :)

|Current | Expected|
|---|---|
|image|image|

**Comment by przbadu** [comment]
Hey, sorry, I was off for few days. Will check it.

**Comment by malparty** [comment]
@przbadu No worries, take you time, we were also quite busy this past weeks and should be back on track by end of next week to follow up with all your PRs ;-)

**Comment by malparty** [comment]
@przbadu just to let you know we have planned a new release for the end of the month. Your previously merged contributions will be included. This PR might not be, unless we can move forward with the feedback ;-)
We will try to have monthly releases to keep the project up to date.
Let me know if I can support you on anything,
Thanks,

**Comment by malparty** [comment]
Closing as long time without activity.

**Pull request closed without merging**